aiohttp/http_writer.py

```diff
@@ -138,29 +139,39 @@ def __init__(self, stream, loop, acquire=True):
         self.output_size = 0
 
         self._eof = False
-        self._buffer = []
         self._compress = None
         self._drain_waiter = None
 
         if self._stream.available:
             self._transport = self._stream.transport
+            self._buffer = None
             self._stream.available = False
         elif acquire:
             self._stream.acquire(self)
 
     def set_transport(self, transport):
         self._transport = transport
 
-        chunk = b''.join(self._buffer)
-        if chunk:
-            transport.write(chunk)
-            self._buffer.clear()
+        if self._buffer is not None:
+            for b in self._buffer:
```

Comment:
if i'm understanding this correctly, the buffer is usually just one item, the headers, so sending them in a for loop or joining won't change much.

The tradeoff however is when the buffer is exactly a small chunk (the headers) and one very large chunk (the body). This will copy the body to create a slightly larger chunk and will be much slower than calling write twice.

I have no idea of how the internals of asyncio works, so I'll of course defer to you here ; but do we have benchmarks for this?